src/qml/editorwidgets/Range.qml

```diff
@@ -7,10 +7,11 @@ EditorWidgetBase {
   id: rangeItem
   enabled: isEnabled
 
+  property bool isDouble: LayerUtils.fieldType(field) === 'double'
   property string widgetStyle: config["Style"] ? config["Style"] : "TextField"
   property int precision: config["Precision"] ? config["Precision"] : 1
-  property real min: config["Min"] !== undefined ? config["Min"] : -Infinity
-  property real max: config["Max"] !== undefined ? config["Max"] : Infinity
+  property real min: config["Min"] !== undefined ? config["Min"] : isDouble ? -Infinity : -2147483647
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER is it available in qml?

Reply:
Not a thing unfortunately